diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index db0e934..d106631 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -57,6 +57,10 @@ function(add_unit_test _tgroup _tname)
             if(Osmium_DEBUG)
                 message("  Adding libs: ${_param_LIBS}")
             endif()
+            if(_param_LIBS MATCHES "lpthread")
+                set_target_properties(${_testid} PROPERTIES COMPILE_FLAGS "-pthread")
+                set_target_properties(${_testid} PROPERTIES LINK_FLAGS "-pthread")
+            endif()
             target_link_libraries(${_testid} ${_param_LIBS})
         endif()
 
diff --git a/test/data-tests/CMakeLists.txt b/test/data-tests/CMakeLists.txt
index a36c31c..de1df9c 100644
--- a/test/data-tests/CMakeLists.txt
+++ b/test/data-tests/CMakeLists.txt
@@ -37,6 +37,8 @@ include_directories("../include")
 #-----------------------------------------------------------------------------
 file(GLOB TESTCASE_CPPS testcases/*.cpp)
 add_executable(testdata-testcases testdata-testcases.cpp ${TESTCASE_CPPS})
+set_target_properties(testdata-testcases PROPERTIES COMPILE_FLAGS "-pthread")
+set_target_properties(testdata-testcases PROPERTIES LINK_FLAGS "-pthread")
 target_link_libraries(testdata-testcases
                       ${OSMIUM_XML_LIBRARIES}
 )
@@ -54,6 +56,8 @@ set_tests_properties(testdata-testcases PROPERTIES
 #
 #-----------------------------------------------------------------------------
 add_executable(testdata-xml testdata-xml.cpp)
+set_target_properties(testdata-xml PROPERTIES COMPILE_FLAGS "-pthread")
+set_target_properties(testdata-xml PROPERTIES LINK_FLAGS "-pthread")
 target_link_libraries(testdata-xml
                       ${OSMIUM_XML_LIBRARIES}
 )
@@ -71,6 +75,8 @@ set_tests_properties(testdata-xml PROPERTIES
 #
 #-----------------------------------------------------------------------------
 add_executable(testdata-overview testdata-overview.cpp)
+set_target_properties(testdata-overview PROPERTIES COMPILE_FLAGS "-pthread")
+set_target_properties(testdata-overview PROPERTIES LINK_FLAGS "-pthread")
 target_link_libraries(testdata-overview
                       ${OSMIUM_XML_LIBRARIES}
                       ${GDAL_LIBRARIES}
@@ -94,6 +100,8 @@ find_program(SPATIALITE spatialite)
 
 if(RUBY AND GEM_json_FOUND AND SPATIALITE)
     add_executable(testdata-multipolygon testdata-multipolygon.cpp)
+    set_target_properties(testdata-multipolygon PROPERTIES COMPILE_FLAGS "-pthread")
+    set_target_properties(testdata-multipolygon PROPERTIES LINK_FLAGS "-pthread")
     target_link_libraries(testdata-multipolygon
                         ${OSMIUM_XML_LIBRARIES}
                         ${GDAL_LIBRARIES}
