commit 8d632398aaff6dc0a8bab48384a2112bbab2e947
Author: Andrew Aladjev <aladjev.andrew@gmail.com>
Date:   Thu Jan 26 14:55:33 2017 +0300

    added ability to toggle doc installation

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 21478ec..f614f38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,6 +60,8 @@ option(INSTALL_UTFCPP    "also install utfcpp headers" OFF)
 
 option(WITH_PROFILING    "add flags needed for profiling" OFF)
 
+option(BUILD_DOC "build documentation" ON)
+
 
 #-----------------------------------------------------------------------------
 #
@@ -353,7 +355,9 @@ if(BUILD_BENCHMARKS)
     add_subdirectory(benchmarks)
 endif()
 
-add_subdirectory(doc)
+if(BUILD_DOC)
+  add_subdirectory(doc)
+endif()
 
 
 #-----------------------------------------------------------------------------
